File: programs/loopInvariants/loopInvariants.java
class loopInvariants{
	public static void main(String args[]){
		int arr[] = {9,87,7,4,7,7789};
		int len = arr.length;
		int smallestSoFar = arr[0];
		int nextIndexToCheck = 1;
		while(nextIndexToCheck!=len){
			if(arr[nextIndexToCheck]<smallestSoFar){
				smallestSoFar = arr[nextIndexToCheck];
			}
			nextIndexToCheck++;
		}
		System.out.println(smallestSoFar);
	}
}

File: programs/loopInvariants/loopInvariants.py
def main():
	arr = [6,5,4,3,2,1]
	smallestSoFar = arr[0]
	nextIndexToCheck = 1;
	length = len(arr)
	while(nextIndexToCheck!=length):
		if arr[nextIndexToCheck]<smallestSoFar:
			smallestSoFar = arr[nextIndexToCheck]
		nextIndexToCheck+=1
	print(smallestSoFar)
if __name__ == '__main__':
	main()
